### ISSUE TITLE
Simple security question

### ISSUE THREAD

**Issue by asakura42** [magic-wormhole/magic-wormhole#428]
What information Wormhole relay knows, what information it keep and how long?

**Comment by piegamesde** [comment]
These are rather vague questions, and the short answer would be "not much; only the usual connection metadata". I recommend browsing [the protocol](https://github.com/magic-wormhole/magic-wormhole-protocols) a bit, maybe starting with the [client server](https://github.com/magic-wormhole/magic-wormhole-protocols/blob/main/server-protocol.md) part.

Also note that there is a distinction between the rendezvous server and a relay server. They have different information.

**Comment by asakura42** [comment]
@piegamesde Thanks. So as I can see, if somebody get logs of relay and mailbox, he couldn't reveal to whom I sent my cat pictures? Only file size and client's "ID"? But I don't understand, is the handshakes protected? I mean when I `wormhole send cat.png` and waiting for receiving, can server admin  see some logs and say "heh, this paranoid from Tokyo sends cat.png and his friend from Florida wants to receive it!" 
Of course admin can anyway see that I connected to his server, but can he bind two participants of file transferring and file details?

**Comment by piegamesde** [comment]
The server always gets to see who communicates with whom, because of IP addresses (there's Tor support built in if you care about that). Similarly, content length metadata is not protected either (because that's a genuinely hard problem and you are usually never protected against that). Otherwise, the first thing clients do before anything else is password negotiation and after that everything is encrypted.

**Comment by asakura42** [comment]
I still think that somebody should describe protocol in simpler terms, like some TL;DR at README. (Maybe I'll try). Anyway thank you for the answers.

**Comment by piegamesde** [comment]
Yeah, I agree. The documentation could be a lot better, on quite a few levels. If some more questions come up, feel free to ask, otherwise please close the issue. Btw we also have #magic-wormhole on both libera.chat (IRC) and matrix.org (Matrix). Feel free to stop by for some quick questions.
